Upgrade CLI to v2 and added arbitrary ranges

mfranzil opened the pull request:
See issue #15.

This PR improves on the tool by:
- updating to Go 1.18,
- updating the urfave/cli package to v2 (with relative changes)
- adding the new Go files to .gitignore,

and most importantly, introduces the possibility of providing a start interval and a duration, rather than a duration which is calculated backwards from the current time. Might need some extra work (as I used a questionable approach for the `steps` function in `prometheus.go`), and probably the vendoring could also be removed.

metalmatze commented on README.md:
````diff
@@ -5,7 +5,7 @@ Export Prometheus data as CSV or directly plot with gnuplot & matplotlib.
 
 ```
-go get -v -u github.com/go-pluto/styx
+go install github.com/mfranzil/styx@latest
````
That needs to be undone.

metalmatze commented on go.mod:
```diff
@@ -0,0 +1,20 @@
+module github.com/mfranzil/styx
```
That needs to be updated to the correct path.

metalmatze commented on go.mod:
```diff
@@ -0,0 +1,20 @@
+module github.com/mfranzil/styx
+
+go 1.18
```
We can already go for 1.20, I guess.

metalmatze commented on prometheus.go:
```diff
@@ -88,13 +88,7 @@ func Query(host string, start time.Time, end time.Time, query string) ([]Result,
 }
 
 func steps(dur time.Duration) int {
```
Hm, this might result in very high amount data, for exporting data with 1w (or even longer) time ranges.

metalmatze reviewed (commented):
Thank you for posting it!
There are a couple of things we should address before merging the PR. Overall it looks really good!

mfranzil commented:
I have fixed the first three issues (and in the meantime, also removed a legacy feature of the old cli v1 API, we can just obtain the parameters without creating structs)

However, I'm kind of unsure how to fix the last one. Should we add options for granularity? We should also probably be more transparent to the user when diminishing it. When I first used the tool, I was confused why having long time ranges had such a low amount of data points.

metalmatze commented:
Yeah, it all depends on the expectation. What we usually do in UI is to have the duration be a pixel on the canvas. For example, given you have 1000 pixel width, for one week we'd do `(7*24*60*60)/1000 ~ 600s = 10m`. Now, this doesn't really work here. We could get the raw output, or have a parameter that specifies how many datapoints someone expects. 

Either way, it seems we can do that one on a follow-up PR.

Thank you so much for these changes!

mfranzil commented:
> Yeah, it all depends on the expectation. What we usually do in UI is to have the duration be a pixel on the canvas. For example, given you have 1000 pixel width, for one week we'd do `(7*24*60*60)/1000 ~ 600s = 10m`. Now, this doesn't really work here. We could get the raw output, or have a parameter that specifies how many datapoints someone expects.
> 
> Either way, it seems we can do that one on a follow-up PR.
> 
> Thank you so much for these changes!

See #18 as a follow-up. You're welcome!
(Also sorry if I respond so late, I tend to procrastinate stuff in my inbox too much ahah)
